src/access.rs

```diff
@@ -0,0 +1,365 @@
+							}
+						}
+					}
+				}
+			}
+			unspent_registered_outputs.push(output);
+		}
+
+		// Sort all confirmed transactions first by block height, then by in-block
+		// position, and finally feed them to the interface in order.
+		confirmed_txs.sort_unstable_by(
+			|(_, block_height1, _, pos1), (_, block_height2, _, pos2)| {
+				block_height1.cmp(&block_height2).then_with(|| pos1.cmp(&pos2))
+			},
+		);
+		for (tx, block_height, block_header, pos) in confirmed_txs {
+			for c in confirmables {
+				c.transactions_confirmed(&block_header, &[(pos, &tx)], block_height);
+			}
+		}
+
+		*self.queued_transactions.lock().unwrap() = Vec::new();
+		*self.watched_transactions.lock().unwrap() = unconfirmed_registered_txs;
+		*self.queued_outputs.lock().unwrap() = Vec::new();
+		*self.watched_outputs.lock().unwrap() = unspent_registered_outputs;
+
+		Ok(())
+	}
+
+	async fn sync_transaction_unconfirmed(
+		&self, confirmables: &Vec<&(dyn Confirm + Sync)>,
+	) -> Result<(), Error> {
+		let client = &*self.blockchain;
+		// Query the interface for relevant txids and check whether they have been
+		// reorged-out of the chain.
+		let relevant_txids =
+			confirmables.iter().flat_map(|c| c.get_relevant_txids()).collect::<HashSet<Txid>>();
+		for txid in relevant_txids {
+			let tx_unconfirmed = client
+				.get_tx_status(&txid)
+				.await
+				.ok()
+				.unwrap_or(None)
+				.map_or(true, |status| !status.confirmed);
+			if tx_unconfirmed {
+				for c in confirmables {
+					c.transaction_unconfirmed(&txid);
+				}
+			}
+		}
+
+		Ok(())
+	}
+
+	pub(crate) fn create_funding_transaction(
+		&self, output_script: &Script, value_sats: u64, confirmation_target: ConfirmationTarget,
+	) -> Result<Transaction, Error> {
+		let num_blocks = num_blocks_from_conf_target(confirmation_target);
+		let fee_rate = self.blockchain.estimate_fee(num_blocks)?;
+
+		let locked_wallet = self.wallet.lock().unwrap();
+		let mut tx_builder = locked_wallet.build_tx();
+
+		tx_builder.add_recipient(output_script.clone(), value_sats).fee_rate(fee_rate).enable_rbf();
+
+		let (mut psbt, _) = tx_builder.finish()?;
+		log_trace!(self.logger, "Created funding PSBT: {:?}", psbt);
+
+		// We double-check that no inputs try to spend non-witness outputs. As we use a SegWit
+		// wallet descriptor this technically shouldn't ever happen, but better safe than sorry.
+		for input in &psbt.inputs {
+			if input.witness_utxo.is_none() {
+				log_error!(self.logger, "Tried to spend a non-witness funding output. This must not ever happen. Panicking!");
+				panic!("Tried to spend a non-witness funding output. This must not ever happen.");
+			}
+		}
+
+		if !locked_wallet.sign(&mut psbt, SignOptions::default())? {
+			return Err(Error::FundingTxCreationFailed);
+		}
+
+		Ok(psbt.extract_tx())
+	}
+
+	pub(crate) fn get_new_address(&self) -> Result<bitcoin::Address, Error> {
+		let address_info = self.wallet.lock().unwrap().get_address(AddressIndex::New)?;
+		Ok(address_info.address)
+	}
+}
```

Comment:
Could these be in a separate commit / PR?

Reply:
Sure, are you fine if I squash the fixups before I break it out to a dedicated commit?

Reply:
SGTM

Reply:
Done in https://github.com/lightningdevkit/ldk-lite/pull/9/commits/d5df62430bc4259e29c5e8ce376d82a48aa148a9